engine_elasticsearch.go

```diff
@@ -164,17 +175,17 @@ func (ee elasticEngine) Ids(collection Collection, stopchan chan struct{}) (chan
 
 func (ee elasticEngine) query(collection Collection, q string, closechan chan struct{}) (chan Document, error) {
 	cont := make(chan Document)
-
 	res, err := ee.client.Post(fmt.Sprintf("%s/%s/%s/_search", ee.baseURL, ee.indexName, collection.name), "application/json", strings.NewReader(q))
 	if err != nil {
 		panic(err)
 	}
-
 	switch {
 	case res.StatusCode == 200:
 	case res.StatusCode == 400:
 		res.Body.Close()
 		return nil, ErrInvalidQuery
+	case res.StatusCode == 404:
+		return nil, ErrNotFound
```

Comment:
If a caller gets "not found" when running a query, would they expect that to mean they their search returned nothing, or that the collection they were searching in doesn't exist?  Maybe it should be explicitly "collection not found" ?